packages/server/src/http/internals/types.ts

```diff
@@ -9,8 +9,31 @@ import { TRPCResponse } from '../../rpc';
 import { Dict } from '../../types';
 import { ResponseMeta } from '../types';
 
+/**
+ * @deprecated use `Headers`
+ */
 export type HTTPHeaders = Dict<string[] | string>;
 
+export function httpHeadersToFetchHeaders(headers: HTTPHeaders): Headers {
+  const newHeaders = new Headers();
+  for (const [key, value] of Object.entries(headers)) {
+    if (!value) {
```

Comment:
I am not sure about this check... what if someone uses the presence of a header rather than its value... people are wierd